test/system/700-play.bats

```diff
@@ -821,3 +821,93 @@ EOF
 
     run_podman rmi $local_image
 }
+
+@test "podman kube play healthcheck should wait initialDelaySeconds before updating status (healthy)" {
+    fname="/tmp/play_kube_healthy_$(random_string 6).yaml"
+    echo "
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+  name: liveness-exec
+spec:
+  containers:
+  - name: liveness
+    image: $IMAGE
+    args:
+    - /bin/sh
+    - -c
+    - touch /tmp/healthy && sleep 100
+    livenessProbe:
+      exec:
+        command:
+        - cat
+        - /tmp/healthy
+      initialDelaySeconds: 3
+      failureThreshold: 1
+      periodSeconds: 1
+" > $fname
+
+    run_podman kube play $fname
+    ctrName="liveness-exec-liveness"
+
+    # wait 2 seconds and check the health status to ensure it is still in
+    # starting state as the initialDelaySeconds is set 3
+    sleep 2
+    run_podman inspect $ctrName --format "{{.State.Health.Status}}"
+    assert $output == "starting"
+    # wait 3 more seconds to give podman enough time to run the health check and
+    # update the status
+    sleep 3
+    run_podman inspect $ctrName --format "{{.State.Health.Status}}"
+    assert $output == "healthy"
+
+    run_podman kube down $fname
+    run_podman pod rm -a
+    run_podman rm -a
+}
+
+@test "podman kube play healthcheck should wait initialDelaySeconds before updating status (unhealthy)" {
+    fname="/tmp/play_kube_unhealthy_$(random_string 6).yaml"
+    echo "
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+  name: liveness-exec
+spec:
+  containers:
+  - name: liveness
+    image: $IMAGE
+    args:
+    - /bin/sh
+    - -c
+    - touch /tmp/healthy && sleep 100
+    livenessProbe:
+      exec:
+        command:
+        - cat
+        - /tmp/randomfile
+      initialDelaySeconds: 3
+      failureThreshold: 1
+      periodSeconds: 1
+" > $fname
+
+    run_podman kube play $fname
+    ctrName="liveness-exec-liveness"
+
+    # wait 2 seconds and check the health status to ensure it is still in
+    # starting state as the initialDelaySeconds is set 3
+    sleep 2
+    run_podman inspect $ctrName --format "{{.State.Health.Status}}"
+    assert $output == "starting"
+    # wait 3 more seconds to give podman enough time to run the health check and
+    # update the status
+    sleep 3
+    run_podman inspect $ctrName --format "{{.State.Health.Status}}"
+    assert $output == "unhealthy"
```

Comment:
This looks fragile, maybe @edsantiago has a better idea here?

Reply:
One possibility (only lightly tested):
```bash
    run_podman kube play $fname

    # Keep checking status. For the first 2 seconds it must be 'starting'
    t0=$SECONDS
    while [[ $SECONDS -le $((t0 + 2)) ]]; do
        run_podman inspect $ctrName --format "1-{{.State.Health.Status}}"
        assert "$output" == "1-starting" "Health.Status at $((SECONDS - t0))"
        sleep 0.5
    done

    # After 2 seconds it may take another second to go unhealthy. Wait.
    t0=$SECONDS
    while [[ $SECONDS -le $((t0 + 2)) ]]; do
        run_podman inspect $ctrName --format "2-{{.State.Health.Status}}"
        if [[ "$output" = "2-unhealthy" ]]; then
            break;
        fi
        sleep 0.5
    done
    assert $output == "2-unhealthy" "After 3 seconds"
```

Reply:
This works! Thanks @edsantiago! 

Fixed